80 update episode objectives (#81)
carpentries-incubator/Natural-language-processing: episodes/03-word_representations.md

::: questions

- What are the steps that matter for defining an NLP task?
- How can words be represented as numbers that capture meaning?
- What kinds of semantic relationships can word embeddings encode?
- What are the limitations of static word representations like Word2Vec?

> - How cab we train our own Word2Vec model?

Reviewer: Typo

```suggestion
- How can we train our own Word2Vec model?
```

::: keypoints

-   An NLP pipeline is a chain of steps from raw text to a structured task output.
-   Word embeddings represent words as dense numeric vectors. These are learned by training neural network(s) on a language modeling objective. Because similar words appear in similar contexts, semantically related words will have gemoetrically similar vectors.
-   Word2Vec is a popular word embedding model. It encodes semantic relationships in vector arithmetic: analogies like "king − man + woman ≈ queen" emerge naturally from the geometry of vectors.

> -   Word2Vec has a key limitation: words unseen during training (out-of-vocabulary) cannot be represented, and each word receives one fixed vector regardless of context.

Reviewer: You actually name two limitations :-)

```suggestion
-   Word2Vec has two key limitations: words that are not in the training data (out-of-vocabulary) cannot be represented, and each word receives one fixed vector regardless of context.
```

-   Cosine similarity measures the angle between two vectors and is the standard metric for comparing word embeddings: it captures semantic relatedness independently of vector magnitude, and returns values between -1 (not similar at all) and 1 (completely similar).
-   Topic modelling is an unsupervised method for discovering latent themes in a document collection, representing each topic as a weighted list of characteristic words and each document as a mixture of topics.

> -   BERTopic builds on language model embeddings by chaining document vectorization (each document gets a single vector that represents it), dimensionality reduction (UMAP),

Reviewer: I wonder if this is too elaborate for an objective. Perhaps merge this point with the previous one, and say something like "... with BERTopic"?
